### PR TITLE
Feat/add distributed tracing support

### PR DESCRIPTION
**Pull request by oodigie**
**Changes:**
- chore: bumped up the packaged ccsmp lib to 7.27.0.1
- feat(SOL-103546): Implement code changes to expose context propagation methods based on ARCH
- test: added unit tests for the new DT functions in the message_impl
- test[SOL-103572]: Golang: Add the integration tests to support the context propagation feature
- test: fixed some failing integration tests
- test: fixed the failing tests in the message impl class

### CODE REVIEW

File: pkg/solace/message/message.go [hunk 1]
```diff
@@ -114,6 +114,38 @@ type Message interface {
...
+	SetTransportTraceContext(traceID [16]byte, spanID [8]byte, sampled bool, traceState string) (ok bool)
+
+	// GetBaggage will return the baggage string associated with the message
+	// If the content is not accessible, an empty slice will
+	// be returned and the ok flag will be false.
+	GetBaggage() (baggage string, ok bool)
+
+	// SetBaggage will set the baggage string associated with the message
+	SetBaggage(baggage string) error
+
```
Comment (mcardy) [inline pkg/solace/message/message.go:148]: I don't think we wanted to expose the interfaces to the world in the Message interface. I think we decided to keep the functions public but hidden to deter users from using these directly. @mgevantmakher can confirm.

File: pkg/solace/message/message.go [hunk 2]
```diff
@@ -114,6 +114,38 @@ type Message interface {
 	//  2 | COS_3
 	GetClassOfService() (cos int)
 
+	// GetCreationTraceContext will return the trace context metadata used for distributed message tracing message
+	// creation context information across service boundaries.
+	// It allows correlating the producer with the consumers of a message, regardless of intermediary
+	// instrumentation. It must not be altered by intermediaries.
+	// If the content is not accessible, an empty slice will be returned and the ok flag will be false.
+	GetCreationTraceContext() (traceID [16]byte, spanID [8]byte, sampled bool, traceState string, ok bool)
```
Comment (mgevantmakher) [inline pkg/solace/message/message.go:122]: all otel methods need to be removed from a public interface

Comment (oodigie) [reply pkg/solace/message/message.go]: these will be removed

**Comment by oodigie** [comment]
The failing test has been updated

**Review by cjwmorgan-sol — COMMENTED** [review]
So far the source looks good to me but I want to compare the tests in solace otel integration project. To ensure nothing is missing.

There is a conflict with dev so that should be resolved as well.

The last jenkins run seems to be having timing issues in the integration tests on linux arm64 along with a test failure on intel linux.
The intel test failure was for test: 

> [Main Suite.[It] DirectReceiver with a messaging service that will be disconnected terminates the receiver using async receive when disconnecting with messaging service disconnect](http://jenkins2/job/solacedev.pubsubplus-go-client/job/feat%252Fadd-distributed-tracing-support/lastCompletedBuild/testReport/(root)/Main%20Suite/linux_x86_64___latest__1_21_4_____It__DirectReceiver_with_a_messaging_service_that_will_be_disconnected_terminates_the_receiver_using_async_receive_when_disconnecting_with_messaging_service_disconnect/)"

Based on the time stamps in the linux arm64 test the initial failure was caused by a known issue in the solace broker where tls truststore updates cause service instabliity in the broker. With the follow tests (based on timestamps) execute immediately afterwards and even smf client connections got interrupted. The broker instability might affect more then just semp services now in the broker. After enough time connections can be established and test start passing again. 

Will continue after I have a better understand the otel field behvaiour from the otel intergration project in pr https://github.com/SolaceDev/pubsubplus-opentelemetry-go-integration/pull/3.

**Comment by oodigie** [comment]
Okay, thanks @cjwmorgan-sol . I have merged changes from dev into this branch and resolved the conflicts. We can investigate the failing integration tests further but I do not think they are related to the changes in this PR.

**Review by cjwmorgan-sol — APPROVED** [review]
Changes look good to me.
The trace state reset logic is implemented in the otel integration component.
All test result are consistent with dev.

This is ready for merge.

**Pull request merged**